keras/engine/training.py

```diff
@@ -2244,6 +2245,17 @@ def save_weights(self,
     if save_format == 'h5':
       with h5py.File(filepath, 'w') as f:
         hdf5_format.save_weights_to_hdf5_group(f, self.layers)
+        weights = self._trainable_weights + self._non_trainable_weights
+        weight_values = backend.batch_get_value(weights)
+        weight_names = [w.name.encode('utf8') for w in weights]
+        hdf5_format.save_attributes_to_hdf5_group(f, 'weight_names', weight_names)
```

Comment:
Importantly, these are not all model weights, just the model's top-level weights (assigned to the model instance directly, not via layers). We should pick a group name that reflects this, e.g. `self_weights`

Reply:
```_trainable_weights``` are top level weights.
You can test:
```python
import tensorflow as tf
import tempfile


class MyModel(tf.keras.Model):
    def __init__(self):
        super(MyModel, self).__init__()
        self.class_token = self.add_weight(shape=[2, 3])
        self.dense = tf.keras.layers.Dense(10)

    def call(self, inputs, training=None, mask=None):
        return self.dense(inputs)


h5_file = tempfile.mktemp('.h5')
m1 = MyModel()
m1.build([None, 10])
print(m1.trainable_weights)
print(m1._trainable_weights)
```